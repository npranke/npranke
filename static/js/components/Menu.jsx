import React from 'react'

function Menu() {
    return (
        <div className="menu">
            <div className="intro">
                Hi, I&#39;m a software engineer; my name is Nicole.
            </div>
            <ul>
                <li>
                    <a href="#" target="_self" rel="noreferrer">
                        small useful things
                    </a>
                </li>
                <li>
                    <a href="#" target="_blank" rel="noreferrer noopener">
                        linkedin
                    </a>
                </li>
            </ul>
        </div>
    )
}

export default Menu
