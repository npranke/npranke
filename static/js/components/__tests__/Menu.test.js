import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import Menu from '../Menu'

Enzyme.configure({ adapter: new Adapter() })

describe('Menu', () => {
    test('has intro text', () => {
        const menu = shallow(<Menu />)

        expect(
            menu.find('.intro').text(),
        ).toBeTruthy()
    })
})

describe('Menu snapshot', () => {
    test('matches snapshot', () => {
        const menu = shallow(<Menu />)

        expect(menu).toMatchSnapshot()
    })
})
