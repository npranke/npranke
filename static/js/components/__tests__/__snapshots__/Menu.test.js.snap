// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshot matches snapshot 1`] = `
<div
  className="menu"
>
  <div
    className="intro"
  >
    Hi, I'm a software engineer; my name is Nicole.
  </div>
  <ul>
    <li>
      <a
        href="#"
        rel="noreferrer"
        target="_self"
      >
        small useful things
      </a>
    </li>
    <li>
      <a
        href="#"
        rel="noreferrer noopener"
        target="_blank"
      >
        linkedin
      </a>
    </li>
  </ul>
</div>
`;
